### PR TITLE
feat(validate): change to use static validatorName instead of instance name

### PR DESCRIPTION
**Pull request by gserb**
Fix for issue #597

### CODE REVIEW

**Comment by daKmoR** [comment]
`static get name` is already used as the actual name/class given by javascript itself and we don't wanna override that

we don't need to store it as we only allow it to be changed by extending and overriding the actual static getter
so in the base class it should look like this

```js
static get validatorName() {
   return ''
}
```

same for async
```js
static get async() {
   return false;
}
```

this means to remove `name` and `async` from the constructor (to remove it from the instance)

Also, we would need to adjust it everywhere where it is used. e.g. for the 4 Validators we have here https://github.com/ing-bank/lion/tree/master/packages/validate/src/validators

this is a breaking change e.g. a lot of tests should/will fail and those will need to be fixed as well.

also, documentation will need to be adjusted 🙈

File: packages/field/test-suites/FormatMixin.suite.js [hunk 1]
```diff
@@ -344,9 +344,13 @@ export function runFormatMixinSuite(customConfig) {
         // that set hasError back to false when the user input is mimicked.
 
         const AlwaysInvalid = class extends Validator {
+          // eslint-disable-next-line no-useless-constructor
           constructor(...args) {
```
Comment (daKmoR) [inline packages/field/test-suites/FormatMixin.suite.js]: eslint is right... let's just remove the constructor here?

File: packages/validate/src/Validator.js [hunk 2]
```diff
@@ -4,22 +4,30 @@ export class Validator {
...
    * @param {string|Date|Number|object} modelValue
    * @param {object} param
    * @returns {Boolean|Promise<Boolean>}
    */
   execute(/* modelValue, param */) {
-    if (!this.name) {
-      throw new Error('You must provide a name like "this.name = \'IsCat\'" for your Validator');
+    if (!this.validatorName) {
+      throw new Error(
+        'You must provide a name like "this.validatorName = \'IsCat\'" for your Validator',
```
Comment (daKmoR) [inline packages/validate/src/Validator.js]:
```suggestion
        A validator needs to have a name! Please set it via "static get validatorName() { return 'IsCat'; }"
```

File: packages/validate/test/DateValidators.test.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ describe('Date Validation', () => {
   it('provides new isDate() to allow only dates', () => {
     let isEnabled;
     const validator = new IsDate();
-    expect(validator.name).to.equal('IsDate');
+    expect(IsDate.validatorName).to.equal('IsDate');
```
Comment (daKmoR) [inline packages/validate/test/DateValidators.test.js]: nit pick 😬 
maybe use `validator.constructor.name`? might be a more realistic use-case 🙈

File: packages/validate/src/Validator.js (hunk 2 above)
Comment (gserb) [reply packages/validate/src/Validator.js]: updated 😊

File: packages/validate/test/DateValidators.test.js (hunk 3 above)
Comment (gserb) [reply packages/validate/test/DateValidators.test.js]: done 😊

File: packages/field/test-suites/FormatMixin.suite.js (hunk 1 above)
Comment (gserb) [reply packages/field/test-suites/FormatMixin.suite.js]: 👌

File: packages/validate/test-suites/ValidateMixin.suite.js [hunk 4]
```diff
@@ -218,9 +221,13 @@ export function runValidateMixinSuite(customConfig) {
 
       it('finally checks for ResultValidators: creates TotalValidationResult', async () => {
         class MyResult extends ResultValidator {
+          // eslint-disable-next-line no-useless-constructor
```
Comment (jorenbroekema) [inline packages/validate/test-suites/ValidateMixin.suite.js]: Remove the constructor

File: packages/validate/test-suites/ValidateMixin.suite.js [hunk 5]
```diff
@@ -391,10 +404,17 @@ export function runValidateMixinSuite(customConfig) {
       });
 
       class IsAsyncCat extends Validator {
+        // eslint-disable-next-line no-useless-constructor
```
Comment (jorenbroekema) [inline packages/validate/test-suites/ValidateMixin.suite.js]: same here

File: packages/input-iban/src/validators.js [hunk 6]
```diff
@@ -36,9 +35,13 @@ export class IsIBAN extends Validator {
 }
 
 export class IsCountryIBAN extends IsIBAN {
+  // eslint-disable-next-line no-useless-constructor
```
Comment (jorenbroekema) [inline packages/input-iban/src/validators.js]: same here

**Comment by jorenbroekema** [comment]
Thanks for your contribution! 🤗

**Pull request merged**